packages/style-engine/src/types.ts

```diff
@@ -0,0 +1,45 @@
+export type Box = {
```

Comment:
Good question, we can probably move most of them over to `CSSProperties` (e.g. `fontWeight`, `fontStyle`, etc). For the `Box` type, since it's used in both margin and padding, it might need to be a bit more flexible than targeting a single CSS property. What do you think would work for that type?

Reply:
Mhh, not sure about that. The main thing is that the idea of `Box` (an object with `top` `right` `bottom` and `left` props) doesn't really translate to CSS.

If we need to maintain the structure of these types, we could still use `CSSProperties` where we're using `string`, and leave the rest unchanged:

```ts
export type Box = {
	top?: CSSProperties[ 'top' ];
	right?: CSSProperties[ 'right' ];
	bottom?: CSSProperties[ 'bottom' ];
	left?: CSSProperties[ 'left' ];
};

export interface Style {
	spacing?: {
		margin?: CSSProperties[ 'margin' ] | Box;
		padding?: CSSProperties[ 'padding' ] | Box;
	};
	// ...
};
```

WDYT?

Reply:
Oh, I quite like that, and it lets us keep the `Box` abstraction. It isn't _quite_ right because it's referring to different properties, but the underlying type should be much the same, and it still serves the purpose of better communicating the desired type. I've updated in https://github.com/WordPress/gutenberg/pull/37978/commits/f0cbeffdc5c0854fa48152e368f1fcfe92871156, and we can continue to tweak the types in follow-ups, too 🙂

Reply:
Thank you! I think it's already much better.

> It isn't quite right because it's referring to different properties, but the underlying type should be much the same,

Out of curiosity, what did you mean with "referring to different properties" ?

Reply:
> what did you mean with "referring to different properties" ?

Apologies, I could have been clearer there! What I meant is that `CSSProperties[ 'top' ]` refers to the [top](https://developer.mozilla.org/en-US/docs/Web/CSS/top) property and not for example [padding-top](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-top) — but since they both accept pretty much the same actual values, it doesn't matter too much that the type isn't precise. I think `CSSProperty[ 'top' ]` is preferable to a more complex type like `CSSProperty[ 'top' ] | CSSProperty[ 'paddingTop ' ] | CSSProperty[ 'marginTop' ]` which could get a bit unwieldy, even if it's technically more accurate.

But, happy to explore in follow-ups, of course!